transitions/select.h

```diff
@@ -0,0 +1,47 @@
+#ifndef TRANSITIONS_SELECT_H
+#define TRANSITIONS_SELECT_H
+
+#include "random.h"
+
+// Usage: TrSelect<SELECTION, TR1, TR2, ...>
+// SELECTION: FUNCTION
+// TR1, TR2: TRANSITION
+// return value: TRANSITION
+// transition option is picked from the specified list of
+// transitions based on Int<>
+// with Int<0> representing first transition
+
+template<class F, class... TRANSITION>
+class TrSelect {
+public:
+  void begin() {
+    begin_ = true;
+  }
+  void run(BladeBase* blade) {
+    f_.run(blade);
+    if (begin_) {
+      begin_ = false;
+      unsigned int n = f_.getInteger(0);
```

Comment:
this won't work right for negative numbers, see what I suggested for ColorSelect<> above.

Reply:
Thanks, I have changed